dailymotion-player-objc/DMPlayerViewController.m

```diff
@@ -30,9 +32,22 @@ - (void)openURLInSafari:(NSURL *)URL;
 @implementation DMPlayerViewController
 
 - (void)dealloc {
-  UIWebView *webView = (UIWebView *)self.view;
-  webView.delegate = nil;
```

Comment:
You're right – well rather put all this into dealloc then. tbh I find that check inside the setter a rather unconventional way of doing it.